docs/add-a-new-datasource.md

```diff
@@ -0,0 +1,70 @@
+
+All resources should be named with the following pattern: `aws_<service>_<name>`
+
+Where `<service>` is the AWS short service name that matches the key in the `serviceData` map in the `conns` package (created via the [Adding a new Service](add-a-new-service.md))
+
+Where `<name>` represents the conceptual infrastructure represented by the create, read, update, and delete methods of the service API. It should be a singular noun. For example, in an API that has methods such as `CreateThing`, `DeleteThing`, `DescribeThing`, and `ModifyThing` the name of the resource would end in `_thing`.
+
+### Fill out the Data Source Schema
+
+In the `internal/service/<service>/<service>.go` file you will see a `Schema` property which exists as a map of `Schema` objects. This relates the AWS API data model with the Terraform resource itself. For each property you want to make available in Terraform, you will need to add it as an attribute, and choose the correct data type.
+
+Attribute names are to specified in `snake_case` as opposed to the AWS API which is `CamelCase`
+
+### Implement Read Handler
+
+These will map the AWS API response to the data source schema. You will also need to handle different response types (including errors correctly). For complex attributes you will need to implement Flattener or Expander functions. The [Data Handling and Conversion Guide](data-handling-and-conversion.md) covers everything you need to know for mapping AWS API responses to Terraform State and vice-versa. The [Error Handling Guide](error-handling.md) covers everything you need to know about handling AWS API responses consistently.
+
+### Write Passing Acceptance Tests
+In order to adequately test the data source we will need to write a complete set of Acceptance Tests. You will need an AWS account for this which allows the provider to read to state of the associated resource. See [Writing Acceptance Tests](running-and-writing-acceptance-tests.md) for a detailed guide on how to approach these.
+
+You will need at minimum:
+
+- Basic Test - Tests full lifecycle (CRUD + Import) of a minimal configuration (all required fields, no optional).
+- Disappears Test - Tests what Terraform does if a resource it is tracking can no longer be found.
+- Per Attribute Tests - For each attribute a test should exists which tests that particular attribute in isolation alongside any required fields.
+
+### Create Documentation for the Data Source
+
+Add a file covering the use of the new resource in `website/docs/d/<service>_<name>.md`. You may want to also add examples of the resource in use particularly if its use is complex, or relies on resources in another service. This documentation will appear on the [Terraform Registry](https://registry.terraform.io/providers/hashicorp/aws/latest) when the resource is made available in a provider release. It is fine to link out to AWS Documentation where appropriate, particularly for values which are likely to change.
```

Comment:
maybe refer to these still as data sources to be consistent with the section header, though i suppose they technically represent aws resources 😅 ? 
```suggestion
Add a file covering the use of the new datasource in `website/docs/d/<service>_<name>.md`. You may want to also add examples of the datasource in use particularly if its use is complex, or relies on resources in another service. This documentation will appear on the [Terraform Registry](https://registry.terraform.io/providers/hashicorp/aws/latest) when the datasource is made available in a provider release. It is fine to link out to AWS Documentation where appropriate, particularly for values which are likely to change.
```